src/utils.rs

```diff
@@ -389,3 +389,48 @@ impl Report {
         });
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #[cfg(feature = "fare-v2")]
+    mod serde_currency {
+        use super::super::*;
+        use serde_derive::{Deserialize, Serialize};
+
+        #[derive(Serialize, Deserialize)]
+        struct CurrencyCodeWrapper {
+            #[serde(
+                serialize_with = "ser_currency_code",
+                deserialize_with = "de_currency_code"
+            )]
+            pub currency_code: String,
+        }
+
+        #[test]
+        fn test_serde_valid_currency_code() {
+            let wrapper = CurrencyCodeWrapper {
+                currency_code: "EUR".to_string(),
+            };
+            let json = serde_json::to_string(&wrapper).unwrap();
+            let wrapper: CurrencyCodeWrapper = serde_json::from_str(&json).unwrap();
+
+            assert_eq!(wrapper.currency_code, "EUR");
+        }
+
+        #[test]
+        fn test_de_invalid_currency_code() {
+            let result: Result<CurrencyCodeWrapper, _> =
+                serde_json::from_str("{\"currency_code\":\"XXX\"}");
+            assert!(result.is_err())
```

Comment:
you don't have better the `is_err`? Can you use `[should_panic]` here?

Reply:
I improved the test to check the error we got is the one we expect (see b13b11d982391e6d78f0d759f52a0e0e9e97f627).

Reply:
I improved the test to check the error we got is the one we expect (see b13b11d982391e6d78f0d759f52a0e0e9e97f627).